src/snowflake/snowpark/session.py

```diff
@@ -695,9 +696,11 @@ def _resolve_imports(
         """Resolve the imports and upload local files (if any) to the stage."""
         resolved_stage_files = []
         stage_file_list = self._list_files_in_stage(
-            stage_location, statement_params=statement_params
+            import_stage, statement_params=statement_params
         )
-        normalized_stage_location = unwrap_stage_location_single_quote(stage_location)
+        # probably shouldn't do it. It is already done in resolve_imports_and_pacakges
+        normalized_import_location = unwrap_stage_location_single_quote(import_stage)
```

Comment:
Can we verify if this could lead to bugs (e.g. over-unwrapping)?

Reply:
there are some tests in `test_utils_suite.py/test_normalize_stage_location` which test for this.

Reply:
Added more tests. Overwrapping does not lead to bugs